api/src/api/article/content-types/article/lifecycles.js

```diff
@@ -0,0 +1,214 @@
+const addOrCreateStage = (entity, newStage) => {
+  if (entity.publicationStages != null) {
+    return {
+      stages: entity.publicationStages.stages.concat(newStage),
+    };
+  } else {
+    return { stages: [newStage] };
+  }
+};
+
+module.exports = {
+  async beforeUpdate(event) {
+    const { params } = event;
+
+    const entity = await strapi.db.query('api::article.article').findOne({
+      where: { id: params.where.id },
+      populate: { createdBy: true },
+    });
+
+    if (!params.populate) {
+      return;
+    }
+    const { publicationState, editors, updatedBy } = params.data;
+
+    let fetchedEditors = [];
+    let fetchedUpdatedBy = {};
+
+    if (editors && editors.length > 0) {
+      try {
+        fetchedEditors = await strapi.db.query('admin::user').findMany({
+          where: {
+            id: {
+              $in: [editors],
+            },
+          },
+          populate: ['email'],
+        });
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    if (updatedBy) {
+      try {
+        fetchedUpdatedBy = await strapi.db.query('admin::user').findOne({
+          where: {
+            id: {
+              $eq: updatedBy,
+            },
+          },
+          populate: ['email'],
+        });
+      } catch (error) {}
+    }
+
+    let message = '';
+    let newPublicationStages = entity.publicationStages;
+    const editorsEmail = fetchedEditors.map((editor) => editor.email);
+
+    // Publication state has changed
+    if (entity.publicationState != publicationState) {
+      // Draft to In review
+      if (
+        publicationState === 'In review' &&
+        entity.publicationState === 'Draft'
+      ) {
+        if (editorsEmail.length > 0) {
+          // try {
```

Comment:
I assume this is commented out because configuring sendmail? if so makes sense but we should include a comment as to why this is commented out.